### DISCUSSION TITLE
Default function parameter values

### DISCUSSION THREAD

**Discussion by RedEmberCat** [vlang/v#12684] [Ideas]
The use of common functions with sane defaults makes interfaces, libraries, and ecosystems easier to use and build. Having to remember every parameter of every interface of every library, or otherwise reference it is needlessly cumbersome. IMO this contradicts the "one way to do things" approach, as it requires the creation of extraneous functions that serve essentially the same purpose.

I wonder if there is a lightweight way to implement this.

**Comment by JalonSolov** [comment]
I'm not sure what you're asking for.  What do you need that generics and short-struct syntax don't handle now?

**Comment by mcastorina** [comment]
Correct me if I'm wrong, but I think the complaint is about various methods doing similar (but not the same) things.

For example, the `os` module has five functions for executing a foreign binary:

```v
fn execute(cmd string) Result
fn execute_or_exit(cmd string) Result
fn execute_or_panic(cmd string) Result
fn execve(cmdpath string, cmdargs []string, envs []string) ?
fn execvp(cmdpath string, cmdargs []string) ?
```

The first three could arguably be reduced to one function via a flag in the parameter to differentiate the behavior. This would make the task of "executing a foreign binary" more "one way" and easier to remember.

**Comment by JalonSolov** [comment]
Well, that's simple enough to fix, but it does change the API, so shouldn't be done lightly.

**Comment by mcastorina** [comment]
This is just one example. Again, correct me if I'm wrong @RedEmberCat . It would be helpful to have a complete list of offenders as well.
